### PR TITLE
final character model animations

### LINKED ISSUE #84: Character Model Asset

### PR DESCRIPTION
**Pull request by emilyyyzh**
Closes #84

### CODE REVIEW

File: Assets/Scripts/PlayerAnimator.cs [hunk 1]
```diff
@@ -8,6 +8,10 @@ public class PlayerAnimator : MonoBehaviour
     public SoundEffect footsteps;
     // For character animation
     public Animator _playerAnimator;
+
+    private string _starting_foot;
+    private bool isWalking = false;
```
Comment (wang-henry) [inline Assets/Scripts/PlayerAnimator.cs:13]:
```suggestion
    private bool _isWalking = false;
```
and below.

File: Assets/Scripts/PlayerAnimator.cs [hunk 2]
```diff
@@ -21,23 +25,44 @@ public void PlayFootSteps()
...
+                _playerAnimator.SetBool(_starting_foot, false);
+                isWalking = false;
+            }
         }
+        
+    }
+
+    bool pressedWalking()
+    {
+        return Input.GetKey(KeyCode.W) || Input.GetKey(KeyCode.S) || Input.GetKey(KeyCode.A) || Input.GetKey(KeyCode.D);
```
Comment (wang-henry) [inline Assets/Scripts/PlayerAnimator.cs:66]:
```suggestion
    private bool pressedWalking()
    {
        return Input.GetKey(KeyCode.W) || Input.GetKey(KeyCode.S) || Input.GetKey(KeyCode.A) || Input.GetKey(KeyCode.D);
```

Comment (wang-henry) [reply Assets/Scripts/PlayerAnimator.cs:66]: Might be better to switch to just detecting movement with the Controller incase we change the controls or allow users to change key binds, maybe something like
```
Vector3 move = new Vector3(Input.GetAxis("Horizontal"), 0, Input.GetAxis("Vertical"));
if(move != Vector3.zero){
..
}
 ```
 Reference: https://docs.unity3d.com/ScriptReference/CharacterController.Move.html

**Comment by wang-henry** [comment]
Also seems to be conflicts with `Assets/Scripts/PlayerAnimator.cs`.

**Review by wang-henry — APPROVED** [review]

File: Assets/Scripts/PlayerAnimator.cs (hunk 2 above)
Comment (ectodrop) [reply Assets/Scripts/PlayerAnimator.cs:66]: @wang-henry @emilyyyzh  we've switched away from the old Unity Input class to using the new Input System which allows us to enable and disable individual controls.

In this case
```
bool pressedWalking()
        return Input.GetKey(KeyCode.W) || Input.GetKey(KeyCode.S) || Input.GetKey(KeyCode.A) || Input.GetKey(KeyCode.D);
```

can be replaced with
```
public GameControls gameControls;
bool pressedWalking()
        return gameControls.Wrapper.Player.Move.IsPressed();
```
check our `CharacterMovement.cs` script for an example on what "GameControls" is, but in short, it's a scriptable object that wraps the new Unity Input System.

**Comment by ectodrop** [comment]
Looks good but just missing the footstep sound effects that were there before, previously they were done with animation events that would call the "PlayFootsteps" function on the `PlayerAnimator.cs` script https://docs.unity3d.com/Manual/script-AnimationWindowEvent.html

**Review by wang-henry — APPROVED** [review]
Looks good to me!

**Pull request merged**